test/instances/steth/StETHHyperdrive.t.sol

```diff
@@ -92,6 +92,100 @@ contract StETHHyperdriveTest is InstanceTest {
             );
     }
 
+    /// @dev Fetches the total supply of the base and share tokens.
+    function getSupply() internal view override returns (uint256, uint256) {
+        return (LIDO.getTotalPooledEther(), LIDO.getTotalShares());
+    }
+
+    /// @dev Fetches the token balance information of an account.
+    function getTokenBalances(
+        address account
+    ) internal view override returns (uint256, uint256) {
+        return (LIDO.sharesOf(account), LIDO.balanceOf(account));
+    }
+
+    /// @dev Verifies that deposit accounting is correct when opening positions.
+    function verifyDeposit(
```

Comment:
Any reason not to abstract this further? It seems like we could define some functions like `getSupply` and `getTokenBalances` that we could use to define this. Maybe I'm missing something, but it seems like the kind of thing that we shouldn't need to implement from scratch for each yield source.

Reply:
I've considered this also, but there is some divergence between what values we assert depending on the instance. The biggest difference is with stETH and the other instances. For example stETH is checking the underlying balance and shares when depositing as shares. With yield bearing tokens it's just a share token balance check. Given the divergence I plan to tackle the lowest hanging fruit first, open/close positions test cases.

Reply:
Ok, that makes sense. As long as we get to this eventually, I'm happy.